File: CPP/Single.cpp
#include<iostream>
using namespace std;

class Base
{
    public :
        int A,B;

        Base()
        {
            cout<<"Inside Base Constructor\n";
        }

        ~Base()
        {
            cout<<"Inside Base Destructor\n";
        }

        void fun()
        {
            cout<<"Inside Base fun\n";
        }
};

class Derived : public Base                         // class Derived extends Base (Syntax of Java)
{
    public :
        int X,Y;

        Derived()
        {
            cout<<"Inside Derived Constructor\n";
        }
        
        ~Derived()
        {
            cout<<"Inside Derived Destructor\n";
        }

        void gun()
        {
            cout<<"Inside Derived gun\n";
        }
};  

int main()
{
    // Derived dobj;                    // Static memory allocation

    Derived * ptr = NULL;

    ptr = new Derived();                // Dynamic memory allocation

    ptr->fun();
    ptr->gun();

    delete ptr;

    return 0;
}

File: Java/Single.java

class Single
{
    public static void main(String arg[])
    {
        //Base bobj1 = new Base();            // No casting
        //Derived dobj1 = new Derived();      // No casting
        Base bobj2 = new Derived();         // Upcasting
        //Derived dobj2 = new Base();       // Downcasting

        bobj2.fun();        // Base fun
        bobj2.fun(11);      // Base fun
        bobj2.gun();        // Derived gun
        //bobj2.sun();        // Derived sun
    }
}

class Base
{
    public int A,B;
    public Base()
    {
        System.out.println("Inside Base Constructor");
        this.A = 10;
        this.B = 20;
    }
    public void fun()       // Definition
    {
        System.out.println("Inside Base Fun");
    }
    public void gun()       // Definition
    {
        System.out.println("Inside Base gun");
    }
    public void fun(int No)     // Overloaded Definition
    {
        System.out.println("Inside Base Fun with one integer");
    }
}

class Derived extends Base      // class Derived : public Base
{
    public int X,Y;
    public Derived()
    {
        System.out.println("Inside Derived Constructor");
        this.X = 30;
        this.Y = 40;   
    }
    public void sun()       // Definition
    {
        System.out.println("Inside Derived sun");
    }
    public void gun()       // Overrided Definition
    {
        System.out.println("Inside Derived gun");
    }
}